[KBJ] First round of cleanups

kevinjacobs-progenity opened the pull request.

nsoranzo committed:
Use cibuildwheel>=1.1.0

to allow specifying the `-L .` option for the auditwheel command.
Introduced in https://github.com/joerick/cibuildwheel/pull/211 .

Fix https://github.com/pysam-developers/pysam/issues/831 .

codeunsolved committed:
Force arg to bytes to support non-ASCII encoding

amblina committed:
Removal of unused import

Import is overwritten in the next line with cy_build

nsoranzo committed:
Split sdist and linux deploy in 3 separate jobs

The Travis job times out after 50'.

rjlasko committed:
Prevent segfault on python AlignedSegment.compare(other=None)

rjlasko committed:
Fix assertion type for unit test comparison

kevinjacobs-progenity committed:
[KBJ] First round of cleanups

AndreasHeger commented:
Thanks, standardization looks good. I can't review in depth, but please feel free to merge.

ExplodingCabbage committed:
Bump minimum Cython version

This is the version @pm67nz indicates is needed for creating a build from source in https://github.com/pysam-developers/pysam/issues/860#issuecomment-572451657, so I figure we might as well enforce it everywhere?

lltw committed:
Fix add_meta function in libcbcf.pyx, so meta-information lines in header added with it have double-quoting rules in accordance to rules specified in VCF4.2 and VCF4.3 specifications

AndreasHeger committed:
Merge pull request #863 from codeunsolved/fix-libcutils-encoding

Force arg to bytes to support non-ASCII encoding

AndreasHeger committed:
Merge pull request #866 from amblina/amblina-patch-1

Removal of unused import

AndreasHeger committed:
Merge pull request #867 from nsoranzo/wheels

Fix wheel building on TravisCI

AndreasHeger committed:
Merge pull request #868 from rjlasko/AlignedSegment-compare-None

Prevent segfault on Python 2.7 AlignedSegment.compare(other=None)

AndreasHeger committed:
Merge pull request #875 from ExplodingCabbage/bump-min-cython

Bump minimum Cython version

AndreasHeger committed:
{AH} add release notes and bump version to 0.15.4

jmarshall committed:
Add Python 3.8 and various devtools/run_tests_travis.sh tidy ups

Add a Python 3.8 build to the Travis build matrix.

Silence wget. No longer need to force use of $PREFIX/{include,lib};
instead ensure we don't see the HTSlib headers in the Conda environment
ahead of pysam's included copy of HTSlib.

Display a more meaningful selection of the HTSlib configuration variables.

nsoranzo committed:
Restore twine upload on tag

AndreasHeger committed:
Merge pull request #878 from nsoranzo/patch-1

Restore twine upload on tag

AndreasHeger committed:
Merge pull request #879 from lltw/fix-add-meta-in-variant-header-class

(issue #881) Fix add_meta function in libcbcf.pyx, so meta-information lines in he…

AndreasHeger committed:
{AH} import utils with package name

kevinjacobs-progenity committed:
[KBJ] Streamline HTSFile and other small cleanups

kevinjacobs-progenity committed:
Merge pysam 0.15.4 into KBJ-cleanup1

graingert committed:
fix region type (PR #883)

jmarshall committed:
Don't use ancient `samtools import` command

The legacy import command was broken in Samtools 1.10 (hat tip @mr-c)
and has been removed from future Samtools (see samtools/samtools#1185).
Use the equivalent `samtools view` command instead.

Update 00README.txt to the current text in samtools/examples/00README.txt
that it was copied from, and delete the samtools command samples as they
are not relevant for pysam.

jmarshall committed:
Update test cases with invalid SAM headers

Several ancient test cases fail with samtools 1.10, as its improved
header facilities require greater baseline validity from input data:

Remove TestDeNovoConstructionUserTags and example_user_header.sam.
These were added in 748ccc8d74dadbfb539b5cbf7e8229edb7228add due
to https://code.google.com/archive/p/pysam/issues/96 but this was
misguided: only `@HD/@SQ/@RG/@PG/@CO` have ever been valid in SAM.

Replace tabs with spaces in rg_with_tab.sam. Samtools now parses
and requires that all tabs do actually delimit header fields, so
pysam's CL-to-end-of-line extension is no longer testable.

Fix whitespace editing error in example_empty_with_header.sam header.

jmarshall committed:
Ensure samtools commands don't add `@PG` headers

Work around the behaviour change in Samtools 1.10 whereby commands
that output SAM/BAM/CRAM now default to adding their own `@PG` headers.
Use GNU Make functions to probe the usage display for the existence
of a --no-PG option, to work with either <= 1.9 or >= 1.10.

jmarshall committed:
Use correct NM values in test data

When converting to CRAM, Samtools 1.10 now preserves incorrect NM values
in its input data, whereas previous versions would effectively correct
them. Change the NM values in ex3.sam to be correct to avoid this
difference between samtools versions, and adjust the SAM/BAM test
cases accordingly.

juliangehring committed:
Fix typos in the FAQ

AndreasHeger committed:
Merge pull request #898 from juliangehring/faq-fix-typos

Fix typos in the FAQ

jmarshall committed:
Merge test suite updates to work with samtools/etc 1.10 (PR #890)

Pysam still bundles htslib/etc 1.9, but with these changes all tests
pass when used together with external samtools/bcftools/etc provided
by either samtools/etc 1.9 or samtools/etc 1.10.x.

jmarshall committed:
Update devtools/import.py and don't import */test/* etc

Pysam's build and tests do not use {htslib,samtools,bcftools}/test
so remove them from the repository and do not import them in future.
Similarly samtools/misc and bcftools/plugins are unused, so remove
them from the import list entirely (rather than listing most misc/*
files and having a note to remove plugins/* manually).

Update so that import.py can be used for htslib too: add the build
infrastructure, but don't run _update_pysam_files() for htslib.

Don't list hfile_irods.c, which was moved to its own repository long
ago and removed from HTSlib 1.4.

jmarshall committed:
Rerun devtools/import.py to remove */misc,plugins,test} etc

Re-import htslib, bcftools, and samtools with the updated script
to remove unused subdirectories and sundries, notably */test.
(Previously the bulk of */test etc were omitted, but various
items had crept back in as this was not fully automated.)

Preserve the hand-made */version.h files and samtools/bamtk.c.pysam.c
fix and the backported htslib/configure fix.

kevinjacobs-progenity committed:
Ensure variant records are unpacked before copying

kevinjacobs-progenity committed:
Merge https://github.com/pysam-developers/pysam into KBJ-cleanup1
